packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -2043,7 +2043,7 @@ class _SliverAppBarState extends State<SliverAppBar> with TickerProviderStateMix
 // center it within its (NavigationToolbar) parent, and allow the
 // parent to constrain the title's actual height.
 class _AppBarTitleBox extends SingleChildRenderObjectWidget {
-  const _AppBarTitleBox({ required Widget super.child });
```

Comment:
With your change, `child` is now nullable, before your change it wasn't. Looking at the code, the expectation seems to be that is is always supposed to be non-null. So, I would consider the old code to be more correct.

Reply:
I now fully understamd why the Widget was explicitly specified